pkg/algorithm/hived_algorithm.go

```diff
@@ -777,10 +839,13 @@ func (h *HivedAlgorithm) processSchedulingRequest(
 	sr schedulingRequest,
```

Comment:
Move all collectNodesNotSuggested (and related log) down to processSchedulingRequest?